tests/fs_additional.rs

```diff
@@ -936,24 +936,44 @@ fn check_metadata(std: &std::fs::Metadata, cap: &cap_std::fs::Metadata) {
 
     // If the standard library supports file modified/accessed/created times,
     // then cap-std should too.
-    if let Ok(expected) = std.modified() {
-        assert_eq!(expected, check!(cap.modified()).into_std());
+    match std.modified() {
+        Ok(expected) => assert_eq!(expected, check!(cap.modified()).into_std()),
+        Err(e) => assert!(
+            cap.modified().is_err(),
+            "modified time should be error ({}), got {:#?}",
+            e,
+            cap.modified()
+        ),
     }
     // The access times might be a little different due to either our own
     // or concurrent accesses.
     const ACCESS_TOLERANCE_SEC: u32 = 60;
-    if let Ok(expected) = std.accessed() {
-        let access_tolerance = std::time::Duration::from_secs(ACCESS_TOLERANCE_SEC.into());
-        assert!(
-            ((expected - access_tolerance)..(expected + access_tolerance))
-                .contains(&check!(cap.accessed()).into_std()),
-            "std accessed {:#?}, cap accessed {:#?}",
-            expected,
+    match std.accessed() {
+        Ok(expected) => {
+            let access_tolerance = std::time::Duration::from_secs(ACCESS_TOLERANCE_SEC.into());
+            assert!(
+                ((expected - access_tolerance)..(expected + access_tolerance))
+                    .contains(&check!(cap.accessed()).into_std()),
+                "std accessed {:#?}, cap accessed {:#?}",
+                expected,
+                cap.accessed()
+            );
+        }
+        Err(e) => assert!(
+            cap.accessed().is_err(),
+            "accessed time should be error ({}), got {:#?}",
+            e,
             cap.accessed()
-        );
+        ),
     }
-    if let Ok(expected) = std.created() {
-        assert_eq!(expected, check!(cap.created()).into_std());
+    match std.created() {
+        Ok(expected) => assert_eq!(expected, check!(cap.created()).into_std()),
+        Err(e) => assert!(
+            cap.created().is_err(),
+            "created time should be error ({}), got {:#?}",
+            e,
+            cap.created()
+        ),
```

Comment:
This test now requires cap-std to match std's behavior exactly (if and only if). Is that a problem? I could imagine one getting support before the other on new operating systems or architectures, but I don't know if failing tests in those cases is a problem.

Reply:
Aha, this failed CI for `musl` with what appears to be a valid timestamp.